core/templates/dev/head/services/AssetsBackendApiService.js

```diff
@@ -37,11 +37,16 @@ oppia.factory('AssetsBackendApiService', [
     var assetsCache = {};
     var _fetchAudio = function(
         explorationId, filename, successCallback, errorCallback) {
-      _filesCurrentlyBeingRequested.push(filename);
+      var canceler = $q.defer();
+      _filesCurrentlyBeingRequested.push({
+        filename: filename,
```

Comment:
This should be a domain object -- FileDownloadRequest or something.

Reply:
Done